core/whistle_media-1.0.0/src/wh_media_map.erl

```diff
@@ -306,9 +306,17 @@ init_map(Db, View, _StartKey, Limit, SendFun, ViewResults) ->
     end.
 
 -spec add_mapping(ne_binary(), fun(), wh_json:objects()) -> 'ok'.
+-spec add_mapping(ne_binary(), fun(), wh_json:objects(), pid()) -> 'ok'.
 add_mapping(Db, SendFun, JObjs) ->
+    add_mapping(Db, SendFun, JObjs, whereis(?MODULE)).
+
+add_mapping(Db, _SendFun, JObjs, Srv) when Srv =:= self() ->
```

Comment:
Since start_link registers `?MODULE` (locally), isn't `Srv` always `self()`?

Note: `whereis/1` "fails" http://erldocs.com/18.2/erts/erlang.html#whereis/1